test/offline/framework/orchestration.cpp

```diff
@@ -0,0 +1,291 @@
+#include "orchestration.h"
+
+#include "execution.h"
+#include "file_management.h"
+
+#include "common/log/log.h"
+#include "common/util/FileUtil.h"
+#include "common/util/StringUtil.h"
+#include "common/util/diff.h"
+
+#include "decompiler/ObjectFile/ObjectFileDB.h"
+#include "test/offline/config/config.h"
+
+#include "third-party/fmt/color.h"
+#include "third-party/fmt/core.h"
+#include "third-party/fmt/ranges.h"
+
+// TODO - this should probably go somewhere common when it's needed eventually
+std::unordered_map<std::string, std::string> game_name_to_config = {
+    {"jak1", "jak1_ntsc_black_label.jsonc"},
+    {"jak2", "jak2_ntsc_v1.jsonc"}};
+
+OfflineTestThreadManager g_offline_test_thread_manager;
+
+OfflineTestDecompiler setup_decompiler(const OfflineTestWorkGroup& work,
+                                       const fs::path& iso_data_path,
+                                       const OfflineTestConfig& offline_config) {
+  // TODO - pull out extractor logic to determine release into common and use here
+  OfflineTestDecompiler dc;
+  dc.config = std::make_unique<decompiler::Config>(
+      decompiler::read_config_file((file_util::get_jak_project_dir() / "decompiler" / "config" /
+                                    game_name_to_config[offline_config.game_name])
+                                       .string()));
+
+  // TODO - do I need to limit the `inputs.jsonc` as well, or is the decompiler smart enough
+  // to lazily load the DGOs as needed based on the allowed objects?
```

Comment:
An outstanding question I had / didn't look into.  Wondering if limiting the `allowed_objects` is sufficient or if I also need to limit the DGO's the decompiler is allowed to use?

Reply:
I think it loads the DGO - it can't do this lazily because it needs to open the dgo to see what's inside.  In theory we could have it trust the `all_objs.json` file to skip dgos that don't contain needed objects, but it's not set up like that right now. It's mostly a leftover for when ObjectFileDB was a tool to unpack what was in the game.

One potential way to fix this is to unpack the DGOs as a pre-processing step (this is fast), then have ObjectFileDB just load object files it needs.